native/src/main/java/io/xlibb/pubsub/PubSub.java

```diff
@@ -62,16 +62,19 @@ public static Object subscribe(Environment environment, BObject pubsub, BString
         } catch (BError bError) {
             return bError;
         }
-        Object[] arguments = new Object[]{timeout, true, typeParam, true};
-        Future future = environment.markAsync();
-        StreamType streamType = TypeCreator.createStreamType(typeParam.getDescribingType(),
-                                                             TypeCreator.createUnionType(PredefinedTypes.TYPE_ERROR,
-                                                                                         PredefinedTypes.TYPE_NULL));
+        Object[] arguments = new Object[]{timeout, typeParam};
+        CompletableFuture<Object> future = new CompletableFuture<>();
         MethodCallback callback = new MethodCallback(future);
-        environment.getRuntime()
-                .invokeMethodAsyncConcurrently(pipe, CONSUME_STREAM_METHOD, null, null, callback, null, streamType,
-                                               arguments);
-        return null;
+        return environment.yieldAndRun(() -> {
```

Comment:
Should we use a new thread here?

Reply:
Fixed with 632ea15b67da0265c250a0b79b2860ba0151ab94